uaclient/entitlements/landscape.py

```diff
@@ -0,0 +1,166 @@
+        )
+        apt.run_apt_install_command([LANDSCAPE_CLIENT_PACKAGE_NAME])
+
+        cmd = ["landscape-config"] + self.extra_args
+        stdout = None
+        stderr = None
+        if self.assume_yes:
+            if "--silent" not in cmd:
+                cmd += ["--silent"]
+            stdout = subprocess.PIPE
+            stderr = subprocess.PIPE
+
+        logging.debug(messages.EXECUTING_COMMAND.format(" ".join(cmd)))
+        event.info(
+            util.redact_sensitive_logs(
+                messages.EXECUTING_COMMAND.format(" ".join(cmd))
+            )
+        )
+        proc = subprocess.run(cmd, stdout=stdout, stderr=stderr)
+        if proc.returncode != 0:
+            if self.assume_yes:
+                stdout_out = proc.stdout.decode("utf-8").strip()
+                stderr_out = proc.stderr.decode("utf-8").strip()
+                err_msg = messages.LANDSCAPE_CONFIG_FAILED
+                event.error(
+                    err_msg.msg,
+                    err_msg.name,
+                    service=self.name,
+                    additional_info={
+                        "stdout": stdout_out,
+                        "stderr": stderr_out,
+                    },
+                )
+                event.info(stderr_out)
+                event.info(
+                    messages.ENABLED_FAILED.format(title=self.title).msg
+                )
+            return False
+
+        event.info(messages.ENABLED_TMPL.format(title=self.title))
+        return True
+
+    def _perform_disable(self, silent: bool = False) -> bool:
+        cmd = ["landscape-config", "--disable"]
+        event.info(messages.EXECUTING_COMMAND.format(" ".join(cmd)))
+        try:
+            system.subp(cmd)
+        except exceptions.ProcessExecutionError as e:
+            with util.disable_log_to_console():
+                logging.error(e)
+            event.info(str(e).strip())
+            event.warning(str(e), self.name)
+
+        msg = messages.BACKING_UP_FILE.format(
+            original=LANDSCAPE_CLIENT_CONFIG_PATH,
+            backup=LANDSCAPE_CLIENT_CONFIG_PATH_DISABLE_BACKUP,
+        )
+        logging.info(msg)
+        event.info(msg)
+        try:
+            os.rename(
+                LANDSCAPE_CLIENT_CONFIG_PATH,
+                LANDSCAPE_CLIENT_CONFIG_PATH_DISABLE_BACKUP,
+            )
+        except FileNotFoundError as e:
+            with util.disable_log_to_console():
+                logging.error(e)
+            event.info(str(e))
+            event.warning(str(e), self.name)
+
+        msg = messages.UNINSTALLING_PACKAGE.format(
+            LANDSCAPE_CLIENT_PACKAGE_NAME
+        )
+        logging.debug(msg)
+        event.info(msg)
+        apt.remove_packages(
+            [LANDSCAPE_CLIENT_PACKAGE_NAME],
+            messages.UNINSTALLING_PACKAGE_FAILED.format(
+                LANDSCAPE_CLIENT_PACKAGE_NAME
+            ),
+        )
```

Comment:
Just for consistency sake, we are removing packages on the other services during disable. Is that a requirement for `Landscape` to do that ?

Reply:
This is what was agreed to in the spec :upside_down_face: